### PR TITLE
fixes for empty chars for pvc prefix

### PR DESCRIPTION
**Pull request by badri-pathak**
## Pull request checklist

```release-notes

```

## Pull request type

This PR is going to fix the issue [Custom Volume Prefix fails for wrong values such as empty string](https://github.com/IBM/ibm-spectrum-scale-csi/issues/1212)
 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

- The current functionality allows empty three char strings also. e.g. "___","a_b","__b"

## What is the new behavior?

- The new fixes will not allow any empty chars in the prefix value. Whenever any wrong or string containing empty chars will pass, it throws an error and default(pvc) will be set as pvc prefix

## How risky is this change?
- [ ] Small, isolated change
- [x] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

**Comment by badri-pathak** [comment]
Image: quay.io/badri_pathak/ibm-spectrum-scale-csi-operator:volNamePrefix_v4

File: operator/controllers/csiscaleoperator_controller.go [hunk 1]
```diff
@@ -2397,11 +2397,11 @@ func validateVolNamePrefix(key string, value string, data map[string]string, inv
 	logger := csiLog.WithName("validateVolNamePrefix")
 	logger.Info("Validating volume name prefix input ", "volNamePrefix", value)
 
-	if len(value) > 2 && len(value) < 6 {
+	if len(value) > 2 && len(value) < 6 && !strings.ContainsAny(value, " ") {
```
Comment (saurabhwani5) [inline operator/controllers/csiscaleoperator_controller.go:2400]: @deeghuge @Jainbrt Here we thinking to add check for special characters as fileset creation fails , please let us know your opinion on same 
`Internal desc = unable to create fileset [$$-2c503dae-585d-4ac9-9c9e-dc04ba92d4cd] in filesystem [fs1]. Error: rpc error: code = Internal desc = remote call failed with response &{{409 Input validation failed: EFSSP1101C CLI: The value "$$-2c503dae-585d-4ac9-9c9e-dc04ba92d4cd" specified for "filesetName" is invalid. The name shall not contain whitespace or any of '#,;:/?&*"()[]'.} []}: POST request`

Comment (badri-pathak) [reply operator/controllers/csiscaleoperator_controller.go:2400]: addressed the comment

**Comment by badri-pathak** [comment]
quay.io/badri_pathak/ibm-spectrum-scale-csi-operator:volNamePrefix_v5

**Comment by saurabhwani5** [comment]
thanks @badri-pathak , 
Image with fixes for upper case of pv name change  :
`quay.io/badri_pathak/ibm-spectrum-scale-csi-operator:volNamePrefix_v6`

**Review by saurabhwani5 — APPROVED** [review]
reviewed and working as expected

**Review by hemalathagajendran — APPROVED** [review]

**Pull request merged**